src/disc_quantile.jl

```diff
@@ -0,0 +1,32 @@
+using Statistics: quantile
+
+struct DiscretizeQuantile <: DiscretizationAlgorithm
+    nbins::Int
+    trim::Bool
+end
+
+# Set trim to false by default
+DiscretizeQuantile(nbins::Int) = DiscretizeQuantile(nbins, false)
+
+function binedges(alg::DiscretizeQuantile, data::AbstractArray{N}) where {N<:Union{Int, AbstractFloat }}
+    nbins = alg.nbins
+
+    n = length(data)
+    n ≥ nbins || error("too many bins requested")
+
+    qs = range(0, 1; length=(nbins+1))
+    retval = quantile.(Ref(data), qs)
+
+    isunique = diff(retval) .> 1e-8
```

Comment:
This is the opposite of is-unique, but rather whether the bin edge is different. Can we do:
```
# trim bins that are too small
keep_bin = diff(retval) .> alg.max_bin_width
```